lib/bap_image/bap_image.ml

```diff
@@ -120,6 +131,8 @@ type t = {
   data : Bigstring.t;
   symbols : symbol table;
   segments : segment table;
+  data_regions : data_region table;
```

Comment:
drop this, it is duplication of the `memory` data structure. Moreover, you already have them in the `spec` entry.